### ISSUE TITLE
Slower than official implementation

### ISSUE THREAD

**Issue by kitckso** [aihacker111/Efficient-Live-Portrait#8]
Tested the ONNX version with 3060, only Animating process:
Official pytorch: 27s
Animating... ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 100% 0:00:27
ONNX: 87s
Animating...: 100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 231/231 [01:27<00:00,  2.63it/s]

Maybe explained by juntaosun
https://github.com/KwaiVGI/LivePortrait/issues/144

**Comment by aihacker111** [comment]
@kitckso Do you run official with compile and triton ?

**Comment by kitckso** [comment]
No, all default setting, only run inference with image and driving video.

**Comment by aihacker111** [comment]
@kitckso Take note, will support for fp16 onnx

**Comment by galigaligo** [comment]
I just https://huggingface.co/myn0908/Live-Portrait-ONNX/tree/main I downloaded the new ONNX and found that it is slower than the original one

**Comment by aihacker111** [comment]
@galigaligo Yes, but that’s not fair for compare onnx with torch, because onnx is design for run on cpu , and when run on gpu it’ll copy the input to gpu device and torch don’t do that, waiting for TensorRT , you can compare it with torch

**Comment by galigaligo** [comment]
> @galigaligo Yes, but that’s not fair for compare onnx with torch, because onnx is design for run on cpu , and when run on gpu it’ll copy the input to gpu device and torch don’t do that, waiting for TensorRT , you can compare it with torch

What I mean is, compared to today's onnx and two days ago's onnx, it's much slower

**Comment by aihacker111** [comment]
@galigaligo because I’m modify some in onnx for correct model that it can convert to tensorrt , if you still use old onnx model, you can’t convert it
